plugins/TwoFactorAuth/Controller.php

```diff
@@ -148,6 +152,13 @@ public function disableTwoFactorAuth()
             $this->twoFa->disable2FAforUser(Piwik::getCurrentUserLogin());
             $this->passwordVerify->forgetVerifiedPassword();
 
+            $container = StaticContainer::getContainer();
+            $email = $container->make(TwoFactorAuthDisabledEmail::class, array(
```

Comment:
Was there an specific reason for using a container->make instead of instantiating the class directly? Would it that way be possible to change the used mail class using DI?

Reply:
I copied the example @diosmosis shared with me. That's the same reason I didn't use `try..catch` for sending the email.

Reply:
This was done in this change: https://github.com/matomo-org/matomo/commit/fcd604b0bc1fdbb4f741a99a58abdc93dd25fd57

It was so plugins could change the contents of the email through DI (I guess we do this on cloud?). I'm not sure if it's needed for these emails (@tsteur would know).

Reply:
it be good if this was possible to change emails generally on cloud 👍